pkg/project/namespace.go

```diff
@@ -43,13 +45,18 @@ func (ns *Namespaced) String() string {
 		return ""
 	}
 
-	var sep string
+	var sep, commitSep, commitID string
 	if ns.IsValid() {
 		sep = "/"
+		if ns.CommitID != nil {
+			commitSep = "#"
+			commitID = ns.CommitID.String()
+		}
 	}
-	return fmt.Sprintf("%s%s%s", ns.Owner, sep, ns.Project)
+	return fmt.Sprintf("%s%s%s%s%s", ns.Owner, sep, ns.Project, commitSep, commitID)
```

Comment:
This is outside of our AC's. It would affect a lot of cases that we're not looking to address with this story. eg. errors that mention the namespace (we don't want to show the commitID there).

Reply:
Yeah, I'm thinking it makes most sense to have a specific type wrapping namespaced to add this behavior if it becomes a more used feature. For now, though, I'll make sure namespaced behaves pretty much exactly as it does before.